pkg/sriov/sriov.go

```diff
@@ -375,57 +375,64 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 	return nil
 }
 
-// ResetVFConfig reset a VF with default values
+// ResetVFConfig reset a VF to its original state
 func (s *sriovManager) ResetVFConfig(conf *sriovtypes.NetConf) error {
 
 	pfLink, err := s.nLink.LinkByName(conf.Master)
 	if err != nil {
 		return fmt.Errorf("failed to lookup master %q: %v", conf.Master, err)
 	}
 
-	// Set vlan to 0
-	if conf.Vlan != 0 {
-		if conf.VlanQoS != 0 {
-			if err = s.nLink.LinkSetVfVlanQos(pfLink, conf.VFID, 0, 0); err != nil {
-				return fmt.Errorf("failed to set vf %d vlan: %v", conf.VFID, err)
+	// Restore VLAN
+	if conf.Vlan != nil {
+		if conf.VlanQoS != nil {
+			if err = s.nLink.LinkSetVfVlanQos(pfLink, conf.VFID, conf.OrigVfState.Vlan, conf.OrigVfState.VlanQoS); err != nil {
+				return fmt.Errorf("failed to restore vf %d vlan: %v", conf.VFID, err)
 			}
-		} else if err = s.nLink.LinkSetVfVlan(pfLink, conf.VFID, 0); err != nil {
-			return fmt.Errorf("failed to set vf %d vlan: %v", conf.VFID, err)
+		} else if err = s.nLink.LinkSetVfVlan(pfLink, conf.VFID, conf.OrigVfState.Vlan); err != nil {
+			return fmt.Errorf("failed to restore vf %d vlan: %v", conf.VFID, err)
+		}
+	}
+
+	// Restore spoofchk
+	if conf.SpoofChk != "" {
```

Comment:
If it restores only when spoofChk is configured, will this fix the issue that spoofChk is flpped to on while releasing VF on mellanox VF?

Reply:
Yes, it is assumed that the VF state prior to cmdAdd() call is a valid one for the NIC driver(so no spoofchk=on, MAC=00000...).
then if user chose to change the value, we will restore to the original state.

Reply:
What if user doesn't specify the value of spoofChk, will it still be turned on automatically when released? (then next time it's used in other pod, traffic blocked)

Reply:
Maybe im missing something, but who would turn on spoofchk automatically ?
the only entity who modifies these VF attributes should be SR-IOV CNI.

Reply:
ok, here is what I found on openshift deployment: when VF is released from pod, spoofChk is turned on automatically, the next time new pod get created with the same VF will have traffic blocked as it doesn't explicitly turn off the spoofChk.

Reply:
> ok, here is what I found on openshift deployment: when VF is released from pod, spoofChk is turned on automatically, the next time new pod get created with the same VF will have traffic blocked as it doesn't explicitly turn off the spoofChk.

Current SR-IOV CNI automatically sets it to on on cmdDel(), after this change it will not.
this PR does not address fixing a VF which has ended up in an invalid state.